File: src/main/java/org/apache/guacamole/requestaccess/service/RequestManagementService.java
package org.apache.guacamole.requestaccess.service;

import com.google.inject.Inject;
import java.util.List;
import org.apache.guacamole.requestaccess.db.RequestMapper;
import org.apache.guacamole.requestaccess.model.Request;

public class RequestManagementService {

    @Inject
    private RequestMapper requestMapper;

    public List<Request> getRequests() {
        return requestMapper.selectAll();
    }

    public void approveRequest(int requestId) {
        requestMapper.updateStatus(requestId, "APPROVED");
    }

    public void denyRequest(int requestId) {
        requestMapper.updateStatus(requestId, "DENIED");
    }

}

File: src/main/resources/frontend/js/requestManagementService.js
angular.module('requestAccess').service('requestManagementService', ['$http', function requestManagementService($http) {

    /**
     * Makes a request to the REST API to get the list of all requests.
     *
     * @returns {Promise} A promise which will resolve with the list of requests.
     */
    this.getRequests = function() {
        return $http.get("api/session/ext/request-access/manage/requests");
    };

    /**
     * Makes a request to the REST API to approve a request.
     *
     * @param {Number} requestId The ID of the request to approve.
     * @returns {Promise} A promise which will resolve when the request is approved.
     */
    this.approveRequest = function(requestId) {
        return $http.put("api/session/ext/request-access/manage/requests/" + requestId + "/approve");
    };

    /**
     * Makes a request to the REST API to deny a request.
     *
     * @param {Number} requestId The ID of the request to deny.
     * @returns {Promise} A promise which will resolve when the request is denied.
     */
    this.denyRequest = function(requestId) {
        return $http.put("api/session/ext/request-access/manage/requests/" + requestId + "/deny");
    };

    /**
     * Makes a request to the REST API to export the list of requests.
     *
     * @param {String} format The format to export the data in (csv or xlsx).
     */
    this.exportRequests = function(format) {
        window.open("api/session/ext/request-access/manage/requests/export/" + format, "_blank");
    };

}]);
